File: Completed/Go/821.go
func shortestToChar(s string, c byte) []int {
    result := make([]int, len(s))
    tmp := 1<<62-1
    
    if s[0] == c {
        tmp = 0
    }
    result[0] = tmp
    
    for i := 1; i < len(s); i++ {
        ch := s[i]
        if ch == c {
            tmp = 0
        } else {
            tmp = result[i-1] +1
        }
        result[i] = tmp
    }
    
    tmp = 1<< 62-1
    if s[len(s)-1] == c {
        tmp = 0
    }
    result[len(s)-1] = min(result[len(s)-1], tmp)
    for i := len(s)-2; i >= 0; i-- {
        ch := s[i]
        if ch == c {
            tmp = 0
        } else {
            tmp = result[i+1]+1
        }
        result[i] = min(result[i], tmp)
    }
    return result
}

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}

File: Completed/python3/821.py
class Solution:
    def shortestToChar(self, s: str, c: str) -> List[int]:
        result = [1<<62] * len(s)
        
        if s[0] == c:
            result[0] = 0
        for i in range(1, len(s), 1):
            ch = s[i]
            if ch == c:
                result[i] = 0
            else:
                result[i] = result[i-1]+1
        
        if s[len(s)-1] == c:
            result[len(s)-1] = 0
        for i in range(len(s)-2, -1, -1):
            ch = s[i]
            if ch == c:
                result[i] = 0
            else:
                result[i] = min(result[i], result[i+1]+1)
        return result
        
